build_prod.py

```diff
@@ -220,6 +220,27 @@ def build_populate_artifacts(cfg):
 
 def generate_local_conf(cfg, reconstruct_dir):
         print('Generating local.conf')
+        cfg_local_conf = cfg.get_opt_local_conf()
+        if not cfg_local_conf:
+            # if no local.conf data - make it empty
+            # the list may be filled from prod local.config
+            cfg_local_conf = []
+        # find the prod local conf
+        local_conf = cfg.get_dir_build() + '//meta-xt-{}//doc//local.conf.{}'.format(cfg.get_opt_product_type(), cfg.get_opt_product_type())
```

Comment:
Not sure we need any logic here at all. Take the local.conf from the product's doc folder and append everything from [local_conf] section

Reply:
in general I agree, what to do if the same variables are defined in both files ?

Reply:
> Not sure we need any logic here at all. Take the local.conf from the product's doc folder and append everything from [local_conf] section

in general I agree, what to do if the same variables are defined in both files ?

Reply:
Then bitbake will take care of it. It is product's and integrator's responsibility to configure the build: product defines what it needs in doc/local.conf and integrator defines his requirements in [local_conf].

Reply:
> Then bitbake will take care of it. It is product's and integrator's responsibility to configure the build: product defines what it needs in doc/local.conf and integrator defines his requirements in [local_conf].

exactly, this is why i would like to have a possibility to override the product configuration for integrator's needs and do not relay on bitbake behavior.

Reply:
and, this code allows to avoid the double-declaration of the variables in the final local.conf. I do not think, that anybody happy with to see the code like
int a=1;
a=2;
...
a=2021;
std::cout << a << std::endl;

is it ok ?

Reply:
I will add two comments to your discussion.

1) Any variable that is appended to end of local.conf will overwrite previous definition. See example here https://docs.yoctoproject.org/bitbake/bitbake-user-manual/bitbake-user-manual-metadata.html#variable-expansion
So, it's OK to have multiple assignments for same variable. And latest read values will be used.

2) I suspect that immediate assignment will be handled incorrectly. I'm talking about following scenario:
We have config for build script with 
```
[local_conf]
VAR_A = 12
```
And we have `doc/local.conf` with text
```
VAR_A := 1
VAR_B := ${A}
```
I suspect that script will overwrite this immediate assignment. But expected behaviour - use immediate assignment of VAR_A for proper assignment of VAR_B, and then use VAR_A from [local_conf] for all other cases.

Also, how we will handle other types of assignments used in product's local conf ( .=  =.  +=  ?= ) ?

Reply:
It's OK to set some variable to different values for different parts of bitbake code. For example we use different ${S} for different places.
Also we can sent variable, call some python function, and set variable to other value. And it's all during configuration stage.

Reply:
1.we are not saying about the behavior, we are saying about the style - useless multiple variable definitions just confuses the reader. 
2.no, script does not override it, because there are different name of the variables
regarding the handling of  ( .= =. += ?= ) - i agree, need to be added

Reply:
> It's OK to set some variable to different values for different parts of bitbake code. For example we use different ${S} for different places.
> Also we can sent variable, call some python function, and set variable to other value. And it's all during configuration stage.

maybe be i did not notice something - sorry for this, but which parts do you mean ? we have a dial just with the local.conf, and it is proposed to have something like
line 1
MYVAR="good"
....
line 101
MYVAR="so bad"

is it good ?